modules/studySearch/src/main/Env.scala

```diff
@@ -16,13 +18,30 @@ final class Env(
 
   val api: StudySearchApi = wire[StudySearchApi]
 
-  def apply(text: String, page: Int)(using me: Option[Me]) =
+  def apply(text: String, order: Order, page: Int)(using me: Option[Me]) =
     Paginator[Study.WithChaptersAndLiked](
       adapter = new AdapterLike[Study]:
-        def query = Query.study(text.take(100), me.map(_.userId.value))
+        def query =
+          Query.study(
+            text.take(100),
+            order.toSpec,
+            me.map(_.userId.value)
+          )
         def nbResults = api.count(query).dmap(_.toInt)
         def slice(offset: Int, length: Int) = api.search(query, From(offset), Size(length))
       .mapFutureList(pager.withChaptersAndLiking()),
       currentPage = page,
       maxPerPage = pager.maxPerPage
     )
+
+  extension (x: Order)
+    def toSpec: Option[StudySorting] =
```

Comment:
sorting is optional in `lila-search`, we probably need to add new `sort` option for study search page.